src/containers/WindowTopBar.js

```diff
@@ -2,7 +2,7 @@ import { compose } from 'redux';
 import { connect } from 'react-redux';
 import * as actions from '../state/actions';
 import miradorWithPlugins from '../lib/miradorWithPlugins';
-import WindowTopBar from '../components/WindowTopBar';
+import TranslatedWindowTopBar from '../components/WindowTopBar';
```

Comment:
Note that we're going to reintroduce the \<ConnectedFooBar/\> pattern here. I see that the problem comes with the `withNamespace()` HOC. At the end of [this wiki article](https://github.com/ProjectMirador/mirador/wiki/M3-Keep-store-connected-components-in-separate-files#problems) I outlined the problem of additional HOCs. In this case I would tend to place the `withNamsepaces()` code in the container rather than the component.

Reply:
I struggled with if i18n was presentational or functional (I can see arguments for both).  Happy to call it functional and put it in the container.

Reply:
@charbugs - is it possible to just change the import statement back to `WindowTopBar` and call it good? I'm a little worried that pushing i18n into the container is going to make it challenging to test if we have to stub out i18n in component tests.

Reply:
You're right, there is no clear distiction. As I said, I'am not sure how to handle this additional-hoc-cases. Perhaps we should generally put HOCs in the containers...